src/main/java/org/broadinstitute/hellbender/utils/test/MiniClusterUtils.java

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.hellbender.utils.test;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+
+public class MiniClusterUtils {
+
+    public static MiniDFSCluster getMiniCluster() throws IOException {
+        final File baseDir = BaseTest.createTempDir("minicluster_storage");
+        final Configuration conf = new Configuration();
+        conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, baseDir.getAbsolutePath());
+        return new MiniDFSCluster.Builder(conf).build();
+    }
+
+    public static Path getTempPath(MiniDFSCluster cluster, String prefix, String extension) throws IOException {
+        return new Path(cluster.getFileSystem().getWorkingDirectory(), prefix + "_" + UUID.randomUUID() + extension);
```

Comment:
If a preference for `createTempFile()`-like approach, below is the source code.  A `long n = random.nextLong();` via `SecureRandom` with a followup double-check for duplicates including an ExponentialRetry should do it:

```Java
        // file name generation
        private static final SecureRandom random = new SecureRandom();
        static File generateFile(String prefix, String suffix, File dir)
            throws IOException
        {
            long n = random.nextLong();
            if (n == Long.MIN_VALUE) {
                n = 0;      // corner case
            } else {
                n = Math.abs(n);
            }

            // Use only the file name from the supplied prefix
            prefix = (new File(prefix)).getName();

            String name = prefix + Long.toString(n) + suffix;
            File f = new File(dir, name);
            if (!name.equals(f.getName()) || f.isInvalid()) {
                if (System.getSecurityManager() != null)
                    throw new IOException("Unable to create temporary file");
                else
                    throw new IOException("Unable to create temporary file, " + f);
            }
            return f;
        }
```